src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -97,8 +93,16 @@ public PooledQueueCache(ICacheDataAdapter<TQueueMessage, TCachedMessage> cacheDa
             this.cacheDataAdapter = cacheDataAdapter;
             this.comparer = comparer;
             this.logger = logger.GetSubLogger("messagecache", "-");
+            this.itemCount = 0;
             pool = new CachedMessagePool<TQueueMessage, TCachedMessage>(cacheDataAdapter);
             messageBlocks = new LinkedList<CachedMessageBlock<TCachedMessage>>();
+            this.cacheMonitor = cacheMonitor;
+
+            if (this.cacheMonitor != null && cacheMonitorWriteInterval.HasValue)
+            {
+                var timer = new Timer(this.ReportCacheMessageStatistics, null, cacheMonitorWriteInterval.Value, cacheMonitorWriteInterval.Value);
```

Comment:
not keeping timer around?  Won't this turn it off when object is garbage collected?  Also, no means of stopping it should the agent responsible for this be shutdown.  May want to consider making the cache disposable and disposing it when receiver is shutdown..  though disposables are annoying :/